reflect.go

```diff
@@ -45,6 +46,14 @@ func (ri *reflectInspector) recordReflection(ssaPkg *ssa.Package) {
 	// if a new reflectAPI is found we need to Re-evaluate all functions which might be using that API
 	newDone := len(ri.result.ReflectAPIs) + len(ri.result.ReflectObjectNames)
 	if newDone > prevDone {
+		// At least one type in this package is used with reflection;
+		// the package name and import path are reachable via reflection as well.
+		// Note that package main is compiled as having import path "main" as well.
+		ri.result.ReflectObjectNames[ri.lpkg.obfuscatedPackageName()] = ri.lpkg.Name
+		if ri.lpkg.Name != "main" {
+			ri.result.ReflectObjectNames[ri.lpkg.obfuscatedImportPath()] = ri.lpkg.ImportPath
+		}
+
```

Comment:
I think this is too broad, reflection is very common I expect this would exclude most import paths of real world packages from obfuscation. 

Maybe instead hide this behavior behind a flag like `-includePaths`, or make it opt-in on a package level via a magic comment/ constant. Or maybe even add the ability to exclude specific packages from obfuscation.

Reply:
Yes, this would lower the quality of obfuscation in some cases, but I strongly believe that the default behavior should be for any Go package to work out of the box. Stronger obfuscation which may break some code should be opt-in, not opt-out. That is better UX - stronger obfuscation is a bit of a power user feature - and should also prevent us from getting dozens more "garble breaks with package X" issues filed when the problem tends to be related to reflection.

So, yes, we can and should add some way to opt into stronger obfuscation even when reflection is used. I would suggest that we tackle that in a follow-up PR because it would be a slightly involved change. I lean towards some sort of directive comment like you suggest.

Reply:
I think this is very similar to giving up on import path obfuscation entirely. 

Other than wails I think there has been no one who's programs break due to import path obfuscation, so I think this is a rare edge case and should be opt-in.

Reply:
Removed for now; see https://github.com/burrowers/garble/pull/904#issuecomment-2585711395.